Change mobile response-list animation and add features to viewing

janakitti opened the pull request:
- Make mobile response-list tab animate with list on open and close
- Add bookmark button to cards view
- Add timestamp to single message view
- Add invalid message count warning

- Adjust z-indices for proper layering of elements and shadows

janakitti committed:
Make mobile hamburger opaque

janakitti committed:
Merge branch 'viewing-component' of https://github.com/virtrolio/virtrolio-site into viewing-component

janakitti committed:
Add bookmark button to cards view

janakitti committed:
Replace delete button with bookmark on carousel view

janakitti committed:
Fix boomark icon path on message-modal

janakitti committed:
Make hamburger tab slide with response-list on mobile

janakitti committed:
Merge branch 'beta-main' into viewing-component

janakitti committed:
Add timestamp to single-message view

janakitti committed:
Add invalid messages warning

janakitti committed:
Reformat code, supress unused css class warnings

ricetech commented on src/app/pages/viewing/viewing.component.html:
```diff
@@ -11,6 +11,16 @@
+          </p>
+        </div>
+        <div *ngIf="invalidMessageCount > 1" class="pt-3">
+          <p class="invalid-msg-warning"> &#9888; {{ invalidMessageCount }} of your messages are not being shown because
+            they do not conform to the message requirements. For more information, <a
+              href="https://virtrolio.web.app/contact">Contact Us</a></p>
```
```suggestion
          <p class="invalid-msg-warning"> &#9888; {{ invalidMessageCount }} of your messages are not being shown because
            they do not conform to the message requirements. <a
              href="https://virtrolio.web.app/contact">Contact Us</a> if you see this message.</p>
```

DabeerAzeez commented on src/app/pages/viewing/messages/messages.component.html:
```diff
@@ -8,7 +8,8 @@
     <div class="col-12">
-      <button (click)="toggleViewStyle()" *ngIf="!viewService.isCarouselView" class="btn-sm btn-primary shadow">Switch to
+      <button (click)="toggleViewStyle()" *ngIf="!viewService.isCarouselView" class="btn-sm btn-primary shadow">Switch
+        to
         Booklet
```
```suggestion
        to Booklet
```

ricetech commented on src/app/pages/viewing/viewing.component.html:
```diff
@@ -11,6 +11,16 @@
         <app-messages [setMessageList]="messageList"></app-messages>
+        <div *ngIf="invalidMessageCount == 1" class="pt-3">
+          <p class="invalid-msg-warning"> &#9888; One of your messages isn't being shown because it does not conform to
+            the message requirements. For more information, <a href="https://virtrolio.web.app/contact">Contact Us</a>
```
```suggestion
            the message requirements. <a href="https://virtrolio.web.app/contact">Contact Us</a>
            if you see this message.
```

DabeerAzeez commented on src/app/pages/viewing/messages/messages.component.ts:
```diff
@@ -58,22 +58,25 @@ export class MessagesComponent implements OnInit {
     const bgB = parseInt(backColor.slice(5), 16);
     let headerTextColor;
     let trashIcon;
+    let bookmarkIcon;
     let popupIcon;
```
```suggestion
    let headerTextColor: string;
    let trashIcon: string;
    let bookmarkIcon: string;
    let popupIcon: string;
```
Change as necessary if the types are wrong.

DabeerAzeez commented on src/app/pages/viewing/responses-list/responses-list.component.css:
```diff
@@ -188,48 +191,108 @@
   }
 }
 
+/*noinspection CssUnusedSymbol*/
```
Comment why it's being suppressed. Same goes for other instances of suppressions here.

DabeerAzeez commented on src/app/pages/viewing/responses-list/responses-list.component.html:
```diff
@@ -37,38 +40,42 @@
+
+    <div class="sidenav-mobile shadow">
+      <a (click)="navToggle()" class="close-btn">&times;</a>
+      <div class="nav-item mb-2 ml-4 response-title">
+        My Messages
+      </div>
```
```suggestion
      <div class="nav-item mb-2 ml-4 response-title">My Messages</div>
```

ricetech commented on src/app/pages/viewing/responses-list/responses-list.component.html:
```diff
@@ -37,38 +40,42 @@
+  <div [class.nav-mobile-open]="navIsOpen" class="side-nav-container-mobile">
+    <div *ngIf="messageList.length !== 0" (click)="navToggle()" [class.flashy-hamburger-mobile]="showNewToViewing"
+         class="hamburger-mobile vertical-center shadow">
+      <img class="chevron" alt="response-list" src="../../../../assets/images/icons/chevron-right.svg">
```
You should animate this chevron in the same fashion as the desktop chevron

ricetech commented on src/app/pages/viewing/responses-list/responses-list.component.html:
```diff
@@ -37,38 +40,42 @@
-        </tr>
-      </table>
+
+    <div class="sidenav-mobile shadow">
```
The z-index of this is too high, it's overriding the message view modal

DabeerAzeez commented on src/app/pages/viewing/responses-list/responses-list.component.ts:
```diff
@@ -18,11 +18,13 @@ export class ResponsesListComponent implements OnInit {
     try {
       this.uid = this.authService.uid();
-    } catch (e) { }
+    } catch (e) {
+    }
```
```suggestion
    } catch (e) {
       AuthService.displayError(e)
    }
```

DabeerAzeez replied:
Check the indentation :P

ricetech reviewed (changes requested):
- Edit the 'You can now Bookmark this page' toast to be not stuck to the bottom

LillianHuang-dev commented on src/app/pages/viewing/responses-list/responses-list.component.css:
```diff
@@ -81,26 +86,24 @@
 }
 
 /* Button component */
 .hamburger-mobile {
```
There is a shadow here next to the arrow that can be seen when the hamburger menu opens (see a bit below and to the right of the X in the picture)
![image](https://user-images.githubusercontent.com/24356367/88462151-2f680e00-ce77-11ea-85fc-04d60dfd281d.png)

janakitti committed:
Apply suggestions from code review

Co-authored-by: Dabeer Abdul-Azeez <59609232+GuyInFridge@users.noreply.github.com>

janakitti committed:
Reword invalid message warning

Co-authored-by: Eric C. <39015447+ricetech@users.noreply.github.com>

janakitti committed:
Comment on css warning supression

janakitti committed:
Merge branch 'viewing-component' of https://github.com/virtrolio/virtrolio-site into viewing-component

janakitti committed:
Flip chevron direction on mobile when opening list

janakitti committed:
Set z-index of hamburger button below that of modal

janakitti committed:
Change z-indices to prevent shadow of hamburger casting onto list

ricetech commented on src/app/pages/viewing/responses-list/responses-list.component.html:
```diff
@@ -37,38 +40,42 @@
+  <div [class.nav-mobile-open]="navIsOpen" class="side-nav-container-mobile">
+    <div *ngIf="messageList.length !== 0" (click)="navToggle()" [class.flashy-hamburger-mobile]="showNewToViewing"
+         class="hamburger-mobile vertical-center shadow">
+      <img *ngIf="!navIsOpen" class="chevron" alt="response-list"
```
z-index is too high, overriding the mobile navbar dropdown

ricetech committed:
Change 'go back to all messages' to 'close'

Done because single messages open in a new tab, so a close button would make more sense

janakitti committed:
Conditionally show Close button on single-message view

janakitti committed:
Increase z-index of navbar

ricetech commented on src/app/pages/viewing/responses-list/responses-list.component.css:
```diff
@@ -74,7 +74,7 @@
   overflow-x: hidden;
   padding-top: 2em;
-  z-index: 1050;
+  z-index: 1050   ;
```
```suggestion
  z-index: 1050;
```

janakitti committed:
Use ngTemplate for single-message Close/Back button

janakitti committed:
Remove unncessary ngIf inside ngTemplate

ricetech reviewed (changes requested):
The initial slide in the carousel is marginally shorter than all of the generated slides

janakitti committed:
Remove unnecessary spacing

Co-authored-by: Eric C. <39015447+ricetech@users.noreply.github.com>

janakitti committed:
Set fixed height of msg-header

janakitti committed:
Merge branch 'viewing-component' of https://github.com/virtrolio/virtrolio-site into viewing-component
